pkg/receiver/mysqlrecordsreceiver/client.go

```diff
@@ -0,0 +1,343 @@
+	if err != nil {
+		logger.Error("failed to create authentication token:", zap.Error(err))
+	}
+	return authenticationToken
+}
+
+//There are 3 scenarios here for creating connection strings for a database connection
+//1. With a plaintext password
+//2. With an encrypted plaintext password
+//3. With an AWS Authentication token to be used as a password
+func newMySQLClient(conf *Config, logger *zap.Logger) client {
+	var port string
+	var basicauthpassword string
+	var connStr string
+	basicauthpassword = conf.Password
+	//Encrypting a plaintext password if a 24 character secret string is provided by the user from an external file
+	if (len(conf.PasswordType) == 0 || conf.PasswordType == "plaintext") && len(conf.EncryptSecretPath) != 0 {
+		MySecret1, err := readMySecret(conf)
+		if err != nil {
+			logger.Error("error in reading encryption secret from file", zap.Error(err))
+		}
+		encText, err := Encrypt(conf.Password, MySecret1, logger)
+		if err != nil {
+			logger.Error("error encrypting your classified text", zap.Error(err))
+		}
+		logger.Debug("The plaintext password can be replaced with this encrypted password.", zap.String("encryptedPassword", encText))
+	}
+	//Decrypting an encrypted password
+	if conf.PasswordType == "encrypted" {
+		MySecret2, err := readMySecret(conf)
+		if err != nil {
+			logger.Error("error in reading encryption secret from file", zap.Error(err))
+		}
+		decText, err := Decrypt(conf.Password, MySecret2, logger)
+		if err != nil {
+			logger.Error("error decrypting your encrypted text: ", zap.Error(err))
+		}
+		basicauthpassword = decText
+	}
+	//Considering default port to be 3306 if not provided by user
+	if len(conf.DBPort) != 0 {
+		port = conf.DBPort
+	} else {
+		logger.Info("dbport empty, considering default 3306")
+		port = "3306"
+	}
+	endpoint := conf.DBHost + ":" + port
+	if conf.AuthenticationMode == "IAMRDSAuth" {
+		authenticationToken := generateIAMAuthToken(endpoint, conf, logger)
+		tlsConf := createIAMRDSTLSConf(conf.AWSCertificatePath, logger)
+		tlserr := mysql.RegisterTLSConfig("custom", &tlsConf)
+		if tlserr != nil {
+			logger.Error("Error %s when RegisterTLSConfig\n", zap.Error(tlserr))
+		}
+		driverConf := mysql.Config{
+			User:                    conf.Username,
+			Passwd:                  authenticationToken,
+			Net:                     conf.Transport,
+			Addr:                    endpoint,
+			DBName:                  conf.Database,
+			AllowNativePasswords:    conf.AllowNativePasswords,
+			TLSConfig:               "custom",
+			AllowCleartextPasswords: true,
+		}
+		connStr = driverConf.FormatDSN()
+	} else {
+		driverConf := mysql.Config{
+			User:                 conf.Username,
+			Passwd:               basicauthpassword,
+			Net:                  conf.Transport,
+			Addr:                 endpoint,
+			DBName:               conf.Database,
+			AllowNativePasswords: conf.AllowNativePasswords,
+		}
+		connStr = driverConf.FormatDSN()
```

Comment:
This line can be move outside of if/else as is the same in both cases

Reply:
Yes.